1.17 Fabric server crashes at startup

When attempting to start the server with DimensionalDoors it crashes with the following error:

```
Exception in thread "main" java.lang.RuntimeException: Failed to setup Fabric server environment! at net.fabricmc.loader.launch.server.FabricServerLauncher.main(FabricServerLauncher.java:58) Caused by: java.lang.RuntimeException: java.lang.RuntimeException: An exception occurred when launching the server! at net.fabricmc.loader.launch.server.FabricServerLauncher.setup(FabricServerLauncher.java:116) at net.fabricmc.loader.launch.server.FabricServerLauncher.main(FabricServerLauncher.java:56) Caused by: java.lang.RuntimeException: An exception occurred when launching the server! at net.fabricmc.loader.launch.server.FabricServerLauncher.launch(FabricServerLauncher.java:70) at net.fabricmc.loader.launch.server.FabricServerLauncher.setup(FabricServerLauncher.java:114) ... 1 more Caused by: java.lang.reflect.InvocationTargetException at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) at java.base/java.lang.reflect.Method.invoke(Method.java:568) at net.fabricmc.loader.launch.server.FabricServerLauncher.launch(FabricServerLauncher.java:68) ... 2 more Caused by: java.lang.RuntimeException: java.lang.reflect.InvocationTargetException at net.fabricmc.loader.game.MinecraftGameProvider.launch(MinecraftGameProvider.java:236) at net.fabricmc.loader.launch.knot.Knot.launch(Knot.java:153) at net.fabricmc.loader.launch.knot.KnotServer.main(KnotServer.java:28) ... 7 more Caused by: java.lang.reflect.InvocationTargetException at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) at java.base/java.lang.reflect.Method.invoke(Method.java:568) at net.fabricmc.loader.game.MinecraftGameProvider.launch(MinecraftGameProvider.java:234) ... 9 more

Caused by: java.lang.NoClassDefFoundError: net/fabricmc/fabric/api/client/render/fluid/v1/FluidRenderHandler at org.dimdev.dimdoors.block.EternalFluidBlock.<init>(EternalFluidBlock.java:25) at org.dimdev.dimdoors.block.ModBlocks.<clinit>(ModBlocks.java:154) at org.dimdev.dimdoors.world.feature.ModFeatures.<clinit>(ModFeatures.java:48) at net.minecraft.class_3864.handler$zzb000$addGateway(class_3864.java:589) at net.minecraft.class_3864.method_17003(class_3864.java:57) at net.minecraft.class_5478.method_30698(class_5478.java:333) at net.minecraft.class_5504.<clinit>(class_5504.java:19) at net.minecraft.class_5458.method_30567(class_5458.java:52) at net.minecraft.class_5458.method_30566(class_5458.java:75) at java.base/java.util.LinkedHashMap.forEach(LinkedHashMap.java:721) at net.minecraft.class_5458.<clinit>(class_5458.java:74) at net.minecraft.class_2378.<clinit>(class_2378.java:266) at net.minecraft.class_2966.method_12851(class_2966.java:44) at net.minecraft.server.Main.main(Main.java:92) ... 14 more

Caused by: java.lang.ClassNotFoundException: net.fabricmc.fabric.api.client.render.fluid.v1.FluidRenderHandler at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641) at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188) at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:520) at net.fabricmc.loader.launch.server.InjectingURLClassLoader.loadClass(InjectingURLClassLoader.java:57) at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:520) at net.fabricmc.loader.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:175) at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:520) ... 28 more
```

Building from the latest changes causes DimDoors to now conflict with the mod HTM v1.1.2 (controls chest/container access privs)

`
[04:27:06] [main/FATAL]: Mixin apply failed htm.mixins.json:events.ExplosionMixin -> net.minecraft.class_1927: org.spongepowered.asm.mixin.injection.throwables.InvalidInjectionException @At("NEW") on net/minecraft/class_1927::HTMExplosionProtectionCheck with priority 1000 cannot inject into net/minecraft/class_1927::method_8350(Z)V merged by org.dimdev.dimdoors.mixin.ExplosionMixin with priority 1000 [PREINJECT Applicator Phase -> htm.mixins.json:events.ExplosionMixin -> Prepare Injections ->  -> handler$zgc000$HTMExplosionProtectionCheck(ZLorg/spongepowered/asm/mixin/injection/callback/CallbackInfo;)V -> Prepare]
org.spongepowered.asm.mixin.injection.throwables.InvalidInjectionException: @At("NEW") on net/minecraft/class_1927::HTMExplosionProtectionCheck with priority 1000 cannot inject into net/minecraft/class_1927::method_8350(Z)V merged by org.dimdev.dimdoors.mixin.ExplosionMixin with priority 1000 [PREINJECT Applicator Phase -> htm.mixins.json:events.ExplosionMixin -> Prepare Injections ->  -> handler$zgc000$HTMExplosionProtectionCheck(ZLorg/spongepowered/asm/mixin/injection/callback/CallbackInfo;)V -> Prepare]
	at org.spongepowered.asm.mixin.injection.code.Injector.findTargetNodes(Injector.java:293) ~[fabric-server-launch.jar:?]
	at org.spongepowered.asm.mixin.injection.code.Injector.find(Injector.java:240) ~[fabric-server-launch.jar:?]
	at org.spongepowered.asm.mixin.injection.struct.InjectionInfo.prepare(InjectionInfo.java:375) ~[fabric-server-launch.jar:?]
	at org.spongepowered.asm.mixin.transformer.MixinTargetContext.prepareInjections(MixinTargetContext.java:1255) ~[fabric-server-launch.jar:?]
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.prepareInjections(MixinApplicatorStandard.java:1033) ~[fabric-server-launch.jar:?]
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.applyMixin(MixinApplicatorStandard.java:388) ~[fabric-server-launch.jar:?]
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.apply(MixinApplicatorStandard.java:320) ~[fabric-server-launch.jar:?]
	at org.spongepowered.asm.mixin.transformer.TargetClassContext.applyMixins(TargetClassContext.java:345) ~[fabric-server-launch.jar:?]
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:569) ~[fabric-server-launch.jar:?]
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:351) ~[fabric-server-launch.jar:?]
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:208) ~[fabric-server-launch.jar:?]
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClassBytes(MixinTransformer.java:178) ~[fabric-server-launch.jar:?]
	at org.spongepowered.asm.mixin.transformer.FabricMixinTransformerProxy.transformClassBytes(FabricMixinTransformerProxy.java:23) ~[fabric-server-launch.jar:?]
	at net.fabricmc.loader.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:162) ~[fabric-server-launch.jar:?]
	at net.fabricmc.loader.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:154) ~[fabric-server-launch.jar:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:520) ~[?:?]
	at com.github.fabricservertools.htm.events.BlockExplodeCallback.lambda$static$1(BlockExplodeCallback.java:14) ~[htm-1.1.2.jar:?]
	at net.fabricmc.fabric.impl.base.event.ArrayBackedEvent.update(ArrayBackedEvent.java:41) ~[94e7c517-a38b-49c4-8c23-c5ca8b1740dc.jar:?]
	at net.fabricmc.fabric.impl.base.event.ArrayBackedEvent.<init>(ArrayBackedEvent.java:37) ~[94e7c517-a38b-49c4-8c23-c5ca8b1740dc.jar:?]
	at net.fabricmc.fabric.impl.base.event.EventFactoryImpl.createArrayBacked(EventFactoryImpl.java:42) ~[94e7c517-a38b-49c4-8c23-c5ca8b1740dc.jar:?]
	at net.fabricmc.fabric.api.event.EventFactory.createArrayBacked(EventFactory.java:61) ~[94e7c517-a38b-49c4-8c23-c5ca8b1740dc.jar:?]
	at com.github.fabricservertools.htm.events.BlockExplodeCallback.<clinit>(BlockExplodeCallback.java:13) ~[htm-1.1.2.jar:?]
	at com.github.fabricservertools.htm.listeners.WorldEventListener.init(WorldEventListener.java:23) ~[htm-1.1.2.jar:?]
	at com.github.fabricservertools.htm.HTM.onInitialize(HTM.java:33) ~[htm-1.1.2.jar:?]
	at net.fabricmc.loader.entrypoint.minecraft.hooks.EntrypointUtils.invoke0(EntrypointUtils.java:47) [fabric-server-launch.jar:?]
	at net.fabricmc.loader.entrypoint.minecraft.hooks.EntrypointUtils.invoke(EntrypointUtils.java:33) [fabric-server-launch.jar:?]
	at net.fabricmc.loader.entrypoint.minecraft.hooks.EntrypointServer.start(EntrypointServer.java:33) [fabric-server-launch.jar:?]
	at net.minecraft.server.Main.main(Main.java:97) [intermediary-server.jar:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:568) ~[?:?]
	at net.fabricmc.loader.game.MinecraftGameProvider.launch(MinecraftGameProvider.java:234) [fabric-server-launch.jar:?]
	at net.fabricmc.loader.launch.knot.Knot.launch(Knot.java:153) [fabric-server-launch.jar:?]
	at net.fabricmc.loader.launch.knot.KnotServer.main(KnotServer.java:28) [fabric-server-launch.jar:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:568) ~[?:?]
	at net.fabricmc.loader.launch.server.FabricServerLauncher.launch(FabricServerLauncher.java:68) [fabric-server-launch.jar:?]
	at net.fabricmc.loader.launch.server.FabricServerLauncher.setup(FabricServerLauncher.java:114) [fabric-server-launch.jar:?]
	at net.fabricmc.loader.launch.server.FabricServerLauncher.main(FabricServerLauncher.java:56) [fabric-server-launch.jar:?]
[04:27:06] [main/INFO]: Loaded door json from /home/amp/.ampdata/instances/CaveMinecraft/Minecraft/./config/dimdoors/doors/iron_dimensional_door.json with id dimdoors:iron_dimensional_door
[04:27:06] [main/INFO]: Loaded door json from /home/amp/.ampdata/instances/CaveMinecraft/Minecraft/./config/dimdoors/doors/quartz_dimensional_door.json with id dimdoors:quartz_dimensional_door
[04:27:06] [main/INFO]: Loaded door json from /home/amp/.ampdata/instances/CaveMinecraft/Minecraft/./config/dimdoors/doors/oak_dimensional_door.json with id dimdoors:oak_dimensional_door
[04:27:06] [main/INFO]: Loaded door json from /home/amp/.ampdata/instances/CaveMinecraft/Minecraft/./config/dimdoors/doors/gold_dimensional_door.json with id dimdoors:gold_dimensional_door
[04:27:07] [main/FATAL]: Failed to start the minecraft server
java.lang.RuntimeException: Could not execute entrypoint stage 'main' due to errors, provided by 'htm'!
	at net.fabricmc.loader.entrypoint.minecraft.hooks.EntrypointUtils.invoke0(EntrypointUtils.java:50) ~[fabric-server-launch.jar:?]
	at net.fabricmc.loader.entrypoint.minecraft.hooks.EntrypointUtils.invoke(EntrypointUtils.java:33) ~[fabric-server-launch.jar:?]
	at net.fabricmc.loader.entrypoint.minecraft.hooks.EntrypointServer.start(EntrypointServer.java:33) ~[fabric-server-launch.jar:?]
	at net.minecraft.server.Main.main(Main.java:97) [intermediary-server.jar:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:568) ~[?:?]
	at net.fabricmc.loader.game.MinecraftGameProvider.launch(MinecraftGameProvider.java:234) [fabric-server-launch.jar:?]
	at net.fabricmc.loader.launch.knot.Knot.launch(Knot.java:153) [fabric-server-launch.jar:?]
	at net.fabricmc.loader.launch.knot.KnotServer.main(KnotServer.java:28) [fabric-server-launch.jar:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:568) ~[?:?]
	at net.fabricmc.loader.launch.server.FabricServerLauncher.launch(FabricServerLauncher.java:68) [fabric-server-launch.jar:?]
	at net.fabricmc.loader.launch.server.FabricServerLauncher.setup(FabricServerLauncher.java:114) [fabric-server-launch.jar:?]
	at net.fabricmc.loader.launch.server.FabricServerLauncher.main(FabricServerLauncher.java:56) [fabric-server-launch.jar:?]
Caused by: org.spongepowered.asm.mixin.transformer.throwables.MixinTransformerError: An unexpected critical error was encountered
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:363) ~[fabric-server-launch.jar:?]
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:208) ~[fabric-server-launch.jar:?]
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClassBytes(MixinTransformer.java:178) ~[fabric-server-launch.jar:?]
	at org.spongepowered.asm.mixin.transformer.FabricMixinTransformerProxy.transformClassBytes(FabricMixinTransformerProxy.java:23) ~[fabric-server-launch.jar:?]
	at net.fabricmc.loader.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:162) ~[fabric-server-launch.jar:?]
	at net.fabricmc.loader.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:154) ~[fabric-server-launch.jar:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:520) ~[?:?]
	at com.github.fabricservertools.htm.events.BlockExplodeCallback.lambda$static$1(BlockExplodeCallback.java:14) ~[htm-1.1.2.jar:?]
	at net.fabricmc.fabric.impl.base.event.ArrayBackedEvent.update(ArrayBackedEvent.java:41) ~[94e7c517-a38b-49c4-8c23-c5ca8b1740dc.jar:?]
	at net.fabricmc.fabric.impl.base.event.ArrayBackedEvent.<init>(ArrayBackedEvent.java:37) ~[94e7c517-a38b-49c4-8c23-c5ca8b1740dc.jar:?]
	at net.fabricmc.fabric.impl.base.event.EventFactoryImpl.createArrayBacked(EventFactoryImpl.java:42) ~[94e7c517-a38b-49c4-8c23-c5ca8b1740dc.jar:?]
	at net.fabricmc.fabric.api.event.EventFactory.createArrayBacked(EventFactory.java:61) ~[94e7c517-a38b-49c4-8c23-c5ca8b1740dc.jar:?]
	at com.github.fabricservertools.htm.events.BlockExplodeCallback.<clinit>(BlockExplodeCallback.java:13) ~[htm-1.1.2.jar:?]
	at com.github.fabricservertools.htm.listeners.WorldEventListener.init(WorldEventListener.java:23) ~[htm-1.1.2.jar:?]
	at com.github.fabricservertools.htm.HTM.onInitialize(HTM.java:33) ~[htm-1.1.2.jar:?]
	at net.fabricmc.loader.entrypoint.minecraft.hooks.EntrypointUtils.invoke0(EntrypointUtils.java:47) ~[fabric-server-launch.jar:?]
	... 17 more
Caused by: org.spongepowered.asm.mixin.throwables.MixinApplyError: Mixin [htm.mixins.json:events.ExplosionMixin] from phase [DEFAULT] in config [htm.mixins.json] FAILED during APPLY
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.handleMixinError(MixinProcessor.java:642) ~[fabric-server-launch.jar:?]
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.handleMixinApplyError(MixinProcessor.java:594) ~[fabric-server-launch.jar:?]
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:356) ~[fabric-server-launch.jar:?]
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:208) ~[fabric-server-launch.jar:?]
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClassBytes(MixinTransformer.java:178) ~[fabric-server-launch.jar:?]
	at org.spongepowered.asm.mixin.transformer.FabricMixinTransformerProxy.transformClassBytes(FabricMixinTransformerProxy.java:23) ~[fabric-server-launch.jar:?]
	at net.fabricmc.loader.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:162) ~[fabric-server-launch.jar:?]
	at net.fabricmc.loader.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:154) ~[fabric-server-launch.jar:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:520) ~[?:?]
	at com.github.fabricservertools.htm.events.BlockExplodeCallback.lambda$static$1(BlockExplodeCallback.java:14) ~[htm-1.1.2.jar:?]
	at net.fabricmc.fabric.impl.base.event.ArrayBackedEvent.update(ArrayBackedEvent.java:41) ~[94e7c517-a38b-49c4-8c23-c5ca8b1740dc.jar:?]
	at net.fabricmc.fabric.impl.base.event.ArrayBackedEvent.<init>(ArrayBackedEvent.java:37) ~[94e7c517-a38b-49c4-8c23-c5ca8b1740dc.jar:?]
	at net.fabricmc.fabric.impl.base.event.EventFactoryImpl.createArrayBacked(EventFactoryImpl.java:42) ~[94e7c517-a38b-49c4-8c23-c5ca8b1740dc.jar:?]
	at net.fabricmc.fabric.api.event.EventFactory.createArrayBacked(EventFactory.java:61) ~[94e7c517-a38b-49c4-8c23-c5ca8b1740dc.jar:?]
	at com.github.fabricservertools.htm.events.BlockExplodeCallback.<clinit>(BlockExplodeCallback.java:13) ~[htm-1.1.2.jar:?]
	at com.github.fabricservertools.htm.listeners.WorldEventListener.init(WorldEventListener.java:23) ~[htm-1.1.2.jar:?]
	at com.github.fabricservertools.htm.HTM.onInitialize(HTM.java:33) ~[htm-1.1.2.jar:?]
	at net.fabricmc.loader.entrypoint.minecraft.hooks.EntrypointUtils.invoke0(EntrypointUtils.java:47) ~[fabric-server-launch.jar:?]
	... 17 more
Caused by: org.spongepowered.asm.mixin.injection.throwables.InvalidInjectionException: @At("NEW") on net/minecraft/class_1927::HTMExplosionProtectionCheck with priority 1000 cannot inject into net/minecraft/class_1927::method_8350(Z)V merged by org.dimdev.dimdoors.mixin.ExplosionMixin with priority 1000 [PREINJECT Applicator Phase -> htm.mixins.json:events.ExplosionMixin -> Prepare Injections ->  -> handler$zgc000$HTMExplosionProtectionCheck(ZLorg/spongepowered/asm/mixin/injection/callback/CallbackInfo;)V -> Prepare]
	at org.spongepowered.asm.mixin.injection.code.Injector.findTargetNodes(Injector.java:293) ~[fabric-server-launch.jar:?]
	at org.spongepowered.asm.mixin.injection.code.Injector.find(Injector.java:240) ~[fabric-server-launch.jar:?]
	at org.spongepowered.asm.mixin.injection.struct.InjectionInfo.prepare(InjectionInfo.java:375) ~[fabric-server-launch.jar:?]
	at org.spongepowered.asm.mixin.transformer.MixinTargetContext.prepareInjections(MixinTargetContext.java:1255) ~[fabric-server-launch.jar:?]
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.prepareInjections(MixinApplicatorStandard.java:1033) ~[fabric-server-launch.jar:?]
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.applyMixin(MixinApplicatorStandard.java:388) ~[fabric-server-launch.jar:?]
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.apply(MixinApplicatorStandard.java:320) ~[fabric-server-launch.jar:?]
	at org.spongepowered.asm.mixin.transformer.TargetClassContext.applyMixins(TargetClassContext.java:345) ~[fabric-server-launch.jar:?]
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:569) ~[fabric-server-launch.jar:?]
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:351) ~[fabric-server-launch.jar:?]
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:208) ~[fabric-server-launch.jar:?]
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClassBytes(MixinTransformer.java:178) ~[fabric-server-launch.jar:?]
	at org.spongepowered.asm.mixin.transformer.FabricMixinTransformerProxy.transformClassBytes(FabricMixinTransformerProxy.java:23) ~[fabric-server-launch.jar:?]
	at net.fabricmc.loader.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:162) ~[fabric-server-launch.jar:?]
	at net.fabricmc.loader.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:154) ~[fabric-server-launch.jar:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:520) ~[?:?]
	at com.github.fabricservertools.htm.events.BlockExplodeCallback.lambda$static$1(BlockExplodeCallback.java:14) ~[htm-1.1.2.jar:?]
	at net.fabricmc.fabric.impl.base.event.ArrayBackedEvent.update(ArrayBackedEvent.java:41) ~[94e7c517-a38b-49c4-8c23-c5ca8b1740dc.jar:?]
	at net.fabricmc.fabric.impl.base.event.ArrayBackedEvent.<init>(ArrayBackedEvent.java:37) ~[94e7c517-a38b-49c4-8c23-c5ca8b1740dc.jar:?]
	at net.fabricmc.fabric.impl.base.event.EventFactoryImpl.createArrayBacked(EventFactoryImpl.java:42) ~[94e7c517-a38b-49c4-8c23-c5ca8b1740dc.jar:?]
	at net.fabricmc.fabric.api.event.EventFactory.createArrayBacked(EventFactory.java:61) ~[94e7c517-a38b-49c4-8c23-c5ca8b1740dc.jar:?]
	at com.github.fabricservertools.htm.events.BlockExplodeCallback.<clinit>(BlockExplodeCallback.java:13) ~[htm-1.1.2.jar:?]
	at com.github.fabricservertools.htm.listeners.WorldEventListener.init(WorldEventListener.java:23) ~[htm-1.1.2.jar:?]
	at com.github.fabricservertools.htm.HTM.onInitialize(HTM.java:33) ~[htm-1.1.2.jar:?]
	at net.fabricmc.loader.entrypoint.minecraft.hooks.EntrypointUtils.invoke0(EntrypointUtils.java:47) ~[fabric-server-launch.jar:?]
	... 17 more
`

--- a/src/main/java/org/dimdev/dimdoors/client/UnderlaidChildItemRenderer.java
+++ b/src/main/java/org/dimdev/dimdoors/client/UnderlaidChildItemRenderer.java
@@ -1,5 +1,7 @@
 package org.dimdev.dimdoors.client;
 
+import net.fabricmc.api.EnvType;
+import net.fabricmc.api.Environment;
 import net.fabricmc.fabric.api.client.rendering.v1.BuiltinItemRendererRegistry;
 import net.minecraft.client.MinecraftClient;
 import net.minecraft.client.render.VertexConsumerProvider;
@@ -8,11 +10,16 @@ import net.minecraft.client.render.model.json.ModelTransformation;
 import net.minecraft.client.util.math.MatrixStack;
 import net.minecraft.item.Item;
 import net.minecraft.item.ItemStack;
+import net.minecraft.item.Items;
 import org.dimdev.dimdoors.item.DimensionalDoorItemRegistrar;
 
+@Environment(EnvType.CLIENT)
 public class UnderlaidChildItemRenderer implements BuiltinItemRendererRegistry.DynamicItemRenderer {
-	private final ItemStack underlay;
 
+	//Dirty hack? IDK this is the only instance of the class, fix later?
+	public final static UnderlaidChildItemRenderer RENDERER = new UnderlaidChildItemRenderer(Items.ENDER_PEARL);
+
+	private final ItemStack underlay;
 	public UnderlaidChildItemRenderer(Item underlay) {
 		this.underlay = new ItemStack(underlay);
 	}

--- a/src/main/java/org/dimdev/dimdoors/fluid/ModFluids.java
+++ b/src/main/java/org/dimdev/dimdoors/fluid/ModFluids.java
@@ -2,6 +2,8 @@ package org.dimdev.dimdoors.fluid;
 
 import java.util.function.Function;
 
+import net.fabricmc.api.EnvType;
+import net.fabricmc.api.Environment;
 import net.minecraft.client.MinecraftClient;
 import net.minecraft.client.texture.Sprite;
 import net.minecraft.fluid.FlowableFluid;
@@ -36,6 +38,7 @@ public class ModFluids {
 		setupFluidRendering(ModFluids.ETERNAL_FLUID, ModFluids.FLOWING_ETERNAL_FLUID, new Identifier("dimdoors:eternal_fluid"));
 	}
 
+	@Environment(EnvType.CLIENT)
 	private static void setupFluidRendering(Fluid still, Fluid flowing, final Identifier textureFluidId) {
 		final Identifier stillSpriteId = new Identifier(textureFluidId.getNamespace(), "block/" + textureFluidId.getPath() + "_still");
 		final Identifier flowingSpriteId = new Identifier(textureFluidId.getNamespace(), "block/" + textureFluidId.getPath() + "_flow");

--- a/src/main/java/org/dimdev/dimdoors/item/DimensionalDoorItemRegistrar.java
+++ b/src/main/java/org/dimdev/dimdoors/item/DimensionalDoorItemRegistrar.java
@@ -1,6 +1,7 @@
 package org.dimdev.dimdoors.item;
 
 import net.fabricmc.api.EnvType;
+import net.fabricmc.api.Environment;
 import net.fabricmc.fabric.api.client.rendering.v1.BuiltinItemRendererRegistry;
 import net.fabricmc.fabric.api.event.registry.RegistryEntryAddedCallback;
 
@@ -39,7 +40,6 @@ import java.util.function.Function;
 
 public class DimensionalDoorItemRegistrar {
 	public static final String PREFIX = "item_ag_dim_";
-	private static final UnderlaidChildItemRenderer renderer = new UnderlaidChildItemRenderer(Items.ENDER_PEARL);
 
 	private final Registry<Item> registry;
 
@@ -118,8 +118,13 @@ public class DimensionalDoorItemRegistrar {
 		}
 		placementFunctions.put(original, dimItem::place);
 		if (FabricLoader.getInstance().getEnvironmentType() == EnvType.CLIENT) {
-			BuiltinItemRendererRegistry.INSTANCE.register(dimItem, renderer);
+			registerItemRenderer(dimItem);
 		}
+	}
+
+	@Environment(EnvType.CLIENT)
+	private void registerItemRenderer(BlockItem dimItem) {
+		BuiltinItemRendererRegistry.INSTANCE.register(dimItem, UnderlaidChildItemRenderer.RENDERER);
 	}
 
 	private static class AutoGenDimensionalDoorItem extends DimensionalDoorItem implements ChildItem {

